crates/rspack_binding_values/src/compilation/mod.rs

```diff
@@ -742,7 +750,10 @@ impl ToNapiValue for JsCompilationWrapper {
           ToNapiValue::to_napi_value(env, r)
         }
         std::collections::hash_map::Entry::Vacant(entry) => {
-          let env_wrapper = Env::from_raw(env);
+          // Leak `Env` to ensure `JsCompilation` being static.
+          // See `JsCompilation::into_instance` for details.
+          let env_wrapper =
+            STATIC_ENV.with(|lock| *lock.get_or_init(|| Box::leak(Box::new(Env::from_raw(env)))));
```

Comment:
Instance's lifetime is bound to `Env`. So I casted `Env` to static here as a workaround. cc @SyMind